credentials/credentials.go

```diff
@@ -55,17 +55,17 @@ var (
 	alpnProtoStr = []string{"h2-14", "h2-15", "h2-16"}
 )
 
-// Credentials defines the common interface all supported credentials must
-// implement.
-type Credentials interface {
-	// GetRequestMetadata gets the current request metadata, refreshing
+// Retriever is an interface that represents types that can
+// retrieve authentication credentials as metadata.
+type Retriever interface {
+	// Metadata gets the current request metadata, refreshing
 	// tokens if required. This should be called by the transport layer on
 	// each request, and the data should be populated in headers or other
 	// context. When supported by the underlying implementation, ctx can
 	// be used for timeout and cancellation.
 	// TODO(zhaoq): Define the set of the qualified keys instead of leaving
 	// it as an arbitrary string.
-	GetRequestMetadata(ctx context.Context) (map[string]string, error)
+	Retrieve(ctx context.Context) (map[string]string, error)
```

Comment:
On Tue, Mar 17, 2015 at 8:53 AM, Burcu Dogan <notifications@github.com>
wrote:

> In credentials/credentials.go
> <https://github.com/grpc/grpc-go/pull/115#discussion_r26585640>:
>
> >  	// tokens if required. This should be called by the transport layer on
> >  	// each request, and the data should be populated in headers or other
> >  	// context. When supported by the underlying implementation, ctx can
> >  	// be used for timeout and cancellation.
> >  	// TODO(zhaoq): Define the set of the qualified keys instead of leaving
> >  	// it as an arbitrary string.
> > -	GetRequestMetadata(ctx context.Context) (map[string]string, error)
> > +	Retrieve(ctx context.Context) (map[string]string, error)
>
>  I am not convinced Retriever is a better name to go
>
> It retrieves credentials, it explains the very basic single action the
> implementers of this interface does. If you're not convinced Retriever is a
> better name, please suggest one. Because Credentials is no way a good name
> for an interface in Go in this context.
>
It does for oauth2 type of protocols but does not for tls/ssh type of
protocols. To be honest, I am okay with "Credentials" but dislike
GetRequestMetadata (I prefer something like AuthToken(). Do you think it
can co-exist with Credentials interface name?). But I used them because
this interface name and member function name are used across various
languages and I do not want the naming for Go off other languages too much
for a similar interface/struct. Also I think "an interface name should
explain the action that is done by the implementers of the interface"
should apply to other languages too.

> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc-go/pull/115/files#r26585640>.
>